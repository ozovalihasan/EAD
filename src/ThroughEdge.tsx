import { EdgeProps, getBezierPath, Node, Position } from 'react-flow-renderer';
import { getEdgeParams } from './utils';
import useStore from './zustand/store';

const ThroughEdge = ({
  id,
  source,
  target,
  style = {},
  label,
  data,
  selected,
}: EdgeProps ) => {
  const { removeEdge } = useStore();

  let sx, sy, tx, ty, sourcePos, targetPos = null
  
  const sourceNode = useStore(store => store.nodes.find( node => node.id === source)) as Node
  const throughNode = useStore(store => store.nodes.find( node => node.id === data.throughNodeId)) as Node
  const targetNode = useStore(store => store.nodes.find( node => node.id === target)) as Node
  const showTextOnEdges = useStore(store => store.showTextOnEdges)
  const mouseOnEdge = useStore(store => store.mouseOnEdgeId ) === id
  
  if (!sourceNode || !throughNode || !targetNode) {
    return <div></div>
  }
  
  ({ sx, sy, tx, ty, sourcePos, targetPos } = getEdgeParams(sourceNode, throughNode))

  const throughEdgePath = getBezierPath({
    sourceX: sx,
    sourceY: sy,
    sourcePosition: sourcePos,
    targetPosition: targetPos,
    targetX: tx,
    targetY: ty,
  });


  ({ sx, sy, tx, ty, sourcePos, targetPos } = getEdgeParams(throughNode, targetNode, label === "through", sourceNode));

  const edgePath = getBezierPath({
    sourceX: sx,
    sourceY: sy,
    sourcePosition: sourcePos,
    targetPosition: targetPos,
    targetX: tx,
    targetY: ty,
  })
  
  
  return (
    <>
      
      <path
        id={`${id}-first`}
        
        style={(selected || mouseOnEdge) ? {stroke: "black", strokeWidth: 3, strokeDasharray: 0} : style}
        className="stroke-first-500 fill-[none] stroke-[1] "
        d={throughEdgePath}
      />

      
      <path
        id={`${id}-second`}
        style={(selected || mouseOnEdge) ? {stroke: "black", strokeWidth: 3, strokeDasharray: 0} : style}
        className="stroke-first-500 fill-[none] stroke-[1] "
        d={edgePath}
      />
      
      
      <path
        id={`${id}-second-wider`}
        className="fill-[none] stroke-[15] stroke-transparent"
        d={edgePath}
      />
      { selected && 
        <foreignObject
          width={40}
          height={40}
          x={(targetPos === Position.Left) ? tx - 40 : tx }
          y={(targetPos === Position.Top) ? ty - 40 : ty }
        >
            <button className="bg-first-500 rounded-full p-2" onClick={() => removeEdge(id)} >
              Ã—
            </button>
        </foreignObject> 
      }
      {showTextOnEdges && 
        <text>
          <textPath href={`#${id}-first`} style={{ fontSize: '12px' }} startOffset="0" >
            {label}
          </textPath>
        </text>
      }
    </>
  );
}

export default ThroughEdge;