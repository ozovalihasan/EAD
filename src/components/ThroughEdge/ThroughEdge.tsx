import { EdgeProps, getBezierPath, Node, Position } from 'react-flow-renderer';
import { RemoveEdgeButton } from '@/components';
import { getEdgeParams } from '@/utils';
import useCustomizationStore from '@/zustandStore/customizationStore';
import useStore from '@/zustandStore/store';

export type ThroughEdgeDataType = {
  throughNodeId: string
}

interface ThroughEdgeType extends Omit<
  EdgeProps, "sourceX" | "sourceY" | "targetX" | "targetY" | "sourcePosition" |"targetPosition"
> {
  data?: ThroughEdgeDataType,
}

export const ThroughEdge = ({
  id,
  source,
  target,
  style = {},
  label,
  data,
  selected,
}: ThroughEdgeType ) => {
  
  let targetX, targetY, targetPosition = null
  let rest = null

  const nodes = useStore(store => store.nodes)
  
  const sourceNode = nodes.find( node => node.id === source) as Node
  const throughNode = nodes.find( node => node.id === (data as ThroughEdgeDataType).throughNodeId) as Node
  const targetNode = nodes.find( node => node.id === target) as Node
  const mouseOnEdge = useStore(store => store.mouseOnEdgeId ) === id
  
  const showTextOnEdges = useCustomizationStore(store => store.showTextOnEdges);

  if (!sourceNode || !throughNode || !targetNode) {
    return <div></div>
  }

  const throughEdgePath = getBezierPath(
    getEdgeParams(sourceNode, throughNode)
  );


  ({ targetX, targetY, targetPosition, ...rest } = getEdgeParams(throughNode, targetNode, label === "through", sourceNode));

  const edgePath = getBezierPath({ targetX, targetY, targetPosition, ...rest })
  
  let orient: string = "0deg" 
  if (targetPosition === Position.Bottom) {
    orient = "180deg"
  } else if (targetPosition === Position.Left) {
    orient = "-90deg"
  } else if (targetPosition === Position.Right) {
    orient = "90deg"
  } else if (targetPosition === Position.Top) {
    orient = "0deg"
  }
  
  return (
    <>
      
      <path
        id={`${id}-first`}
        style={(selected || mouseOnEdge) ? {stroke: "black", strokeWidth: 3, strokeDasharray: 0} : style}
        className="stroke-first-500 fill-[none] stroke-[1] custom-animation"
        d={throughEdgePath}
      />

      
      <defs>
        <marker   
          orient={orient}
          id={`through-arrow-${id}` }
          viewBox="0 0 200 200"
          refX="100" 
          refY="0" 
          markerUnits="strokeWidth" 
          markerWidth="10px" 
          markerHeight="10px"
          strokeLinecap="round"
        >
          <path 
            className="stroke-[15] stroke-first-500 fill-transparent"
            style={{strokeLinecap:"round", strokeLinejoin: "round"}} 
            d="M 20,20 100,180 180,20 M 20,20 H 180"
          />
        </marker>
      </defs>
      
      <path
        id={`${id}-second`}
        style={(selected || mouseOnEdge) ? {stroke: "black", strokeWidth: 3, strokeDasharray: 0} : style}
        className="stroke-first-500 fill-[none] stroke-[1] custom-animation"
        d={edgePath}
        markerEnd={`url(#through-arrow-${id})`}
      />
      
      <path
        id={`${id}-second-wider`}
        className="fill-[none] stroke-[15] stroke-transparent"
        d={edgePath}
      />
      
      { selected && 
        <foreignObject
          className='h-10 w-10'
          x={(targetPosition === Position.Left) ? targetX - 40 : targetX }
          y={(targetPosition === Position.Top) ? targetY - 40 : targetY }
        >
            <RemoveEdgeButton edgeId={id}/>
        </foreignObject> 
      }
      {showTextOnEdges &&
        <text>
          <textPath href={`#${id}-first`} style={{ fontSize: '12px' }} startOffset="0" >
            {label}
          </textPath>
        </text>
      }
    </>
  );
}