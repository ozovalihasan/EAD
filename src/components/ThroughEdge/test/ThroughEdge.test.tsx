import { ThroughEdge } from '../ThroughEdge';
import { render, screen } from "@testing-library/react";
import useStore from '@/zustandStore/store';
import useCustomizationStore from '@/zustandStore/customizationStore';


let renderReadyComponent: JSX.Element;

beforeEach(() => {
  
  useStore.setState({ 
    nodes: [
      {
        "id": "1",
        "type": "entity",
        "position": {
            "x": 465,
            "y": 150
        },
        "data": {
            "tableId": "1",
            "name": "Physician"
        },
        "width": 138,
        "height": 45
      },
      {
          "id": "2",
          "type": "entity",
          "position": {
              "x": 585,
              "y": 255
          },
          "data": {
              "tableId": "2",
              "name": "Appointment"
          },
          "width": 138,
          "height": 45,
          "selected": false,
          "positionAbsolute": {
              "x": 585,
              "y": 255
          },
          "dragging": false
      },
      {
          "id": "3",
          "type": "entity",
          "position": {
              "x": 720,
              "y": 150
          },
          "data": {
              "tableId": "3",
              "name": "Patient"
          },
          "width": 138,
          "height": 45,
          "selected": false,
          "positionAbsolute": {
              "x": 720,
              "y": 150
          },
          "dragging": false
      }
    ]
  })
  
  renderReadyComponent = (
    <svg>
      <ThroughEdge
        id= {"111"}
        source= {"1"}
        target= {"3"}
        label= {"through"}
        data= {{throughNodeId: "2"}}
        selected={false}
      />
    </svg>
  );
});

describe('<ThroughEdge />', () => {
  describe("if the mouse hovers on the edge", () => {
    beforeEach(()=> {
      useStore.setState({ 
        mouseOnEdgeId: "111"
      })
    })

    it('update styles ', () => {
      
      render(renderReadyComponent );

      const allPathElements = screen.getAllByText("")[0].getElementsByTagName('path')

      expect(allPathElements[0]).toHaveStyle({stroke: "black", strokeWidth: 3, strokeDasharray: 0});
      expect(allPathElements[2]).toHaveStyle({stroke: "black", strokeWidth: 3, strokeDasharray: 0});
      expect(allPathElements.length).toBe(4);
    });
  
     it('renders correctly', () => {
      const renderedContainer = render(renderReadyComponent );
      expect(renderedContainer).toMatchSnapshot();
    });
    
  })
  
  describe("if texts are shown on the edges", () => {
    beforeEach(()=> {
      useCustomizationStore.setState({ 
        showTextOnEdges: true
      })
    })

    it('shows the text ', () => {
      
      render(renderReadyComponent );
      expect(screen.getByText(/through/i)).toBeInTheDocument();
    });
  
    it('renders correctly', () => {
      const renderedContainer = render(renderReadyComponent );
      expect(renderedContainer).toMatchSnapshot();
    });
    
  })

  describe("if it is selected", () => {
    beforeEach(()=> {
      renderReadyComponent =(
        <svg>
          <ThroughEdge
            id= {"111"}
            source= {"1"}
            target= {"3"}
            label= {"through"}
            data= {{throughNodeId: "2"}}
            selected={true}
          />
        </svg>
      )
    })

    it('renders another component', () => {
      
      render(renderReadyComponent );
      expect(screen.getByText(/MockRemoveEdgeButton/i)).toBeInTheDocument();
    });
  
    it('renders correctly', () => {
      const renderedContainer = render(renderReadyComponent );
      expect(renderedContainer).toMatchSnapshot();
    });
    
  })

});