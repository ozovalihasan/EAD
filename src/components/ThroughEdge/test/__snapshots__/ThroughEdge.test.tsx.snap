// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ThroughEdge /> if it is selected renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <svg>
        <path
          class="stroke-first-500 fill-[none] stroke-[1] "
          d="M559.7142857142857,205 C559.7142857142857,225 628.2857142857143,225 628.2857142857143,245"
          id="111-first"
          style="stroke: black; stroke-width: 3; stroke-dasharray: 0;"
        />
        <defs>
          <marker
            id="through-arrow-111"
            markerHeight="10px"
            markerUnits="strokeWidth"
            markerWidth="10px"
            orient="180deg"
            refX="100"
            refY="0"
            viewBox="0 0 200 200"
          >
            <path
              class="fill-first-500"
              d="M 200,-0.34918858 100,199.65081 0,-0.34918858 Z"
            />
          </marker>
        </defs>
        <path
          class="stroke-first-500 fill-[none] stroke-[1] custom-animation"
          d="M628.2857142857143,245 C628.2857142857143,225 789.6428571428571,225 789.6428571428571,205"
          id="111-second"
          marker-end="url(#through-arrow-111)"
          style="stroke: black; stroke-width: 3; stroke-dasharray: 0;"
        />
        <path
          class="fill-[none] stroke-[15] stroke-transparent"
          d="M628.2857142857143,245 C628.2857142857143,225 789.6428571428571,225 789.6428571428571,205"
          id="111-second-wider"
        />
        <foreignobject
          class="h-4 w-4 border-none rounded-full relative"
          x="789.6428571428571"
          y="205"
        >
          MockRemoveEdgeButton edgeId: 
          111
        </foreignobject>
      </svg>
    </div>
  </body>,
  "container": <div>
    <svg>
      <path
        class="stroke-first-500 fill-[none] stroke-[1] "
        d="M559.7142857142857,205 C559.7142857142857,225 628.2857142857143,225 628.2857142857143,245"
        id="111-first"
        style="stroke: black; stroke-width: 3; stroke-dasharray: 0;"
      />
      <defs>
        <marker
          id="through-arrow-111"
          markerHeight="10px"
          markerUnits="strokeWidth"
          markerWidth="10px"
          orient="180deg"
          refX="100"
          refY="0"
          viewBox="0 0 200 200"
        >
          <path
            class="fill-first-500"
            d="M 200,-0.34918858 100,199.65081 0,-0.34918858 Z"
          />
        </marker>
      </defs>
      <path
        class="stroke-first-500 fill-[none] stroke-[1] custom-animation"
        d="M628.2857142857143,245 C628.2857142857143,225 789.6428571428571,225 789.6428571428571,205"
        id="111-second"
        marker-end="url(#through-arrow-111)"
        style="stroke: black; stroke-width: 3; stroke-dasharray: 0;"
      />
      <path
        class="fill-[none] stroke-[15] stroke-transparent"
        d="M628.2857142857143,245 C628.2857142857143,225 789.6428571428571,225 789.6428571428571,205"
        id="111-second-wider"
      />
      <foreignobject
        class="h-4 w-4 border-none rounded-full relative"
        x="789.6428571428571"
        y="205"
      >
        MockRemoveEdgeButton edgeId: 
        111
      </foreignobject>
    </svg>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`<ThroughEdge /> if texts are shown on the edges renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <svg>
        <path
          class="stroke-first-500 fill-[none] stroke-[1] "
          d="M559.7142857142857,205 C559.7142857142857,225 628.2857142857143,225 628.2857142857143,245"
          id="111-first"
        />
        <defs>
          <marker
            id="through-arrow-111"
            markerHeight="10px"
            markerUnits="strokeWidth"
            markerWidth="10px"
            orient="180deg"
            refX="100"
            refY="0"
            viewBox="0 0 200 200"
          >
            <path
              class="fill-first-500"
              d="M 200,-0.34918858 100,199.65081 0,-0.34918858 Z"
            />
          </marker>
        </defs>
        <path
          class="stroke-first-500 fill-[none] stroke-[1] custom-animation"
          d="M628.2857142857143,245 C628.2857142857143,225 789.6428571428571,225 789.6428571428571,205"
          id="111-second"
          marker-end="url(#through-arrow-111)"
        />
        <path
          class="fill-[none] stroke-[15] stroke-transparent"
          d="M628.2857142857143,245 C628.2857142857143,225 789.6428571428571,225 789.6428571428571,205"
          id="111-second-wider"
        />
        <text>
          <textpath
            href="#111-first"
            startOffset="0"
            style="font-size: 12px;"
          >
            through
          </textpath>
        </text>
      </svg>
    </div>
  </body>,
  "container": <div>
    <svg>
      <path
        class="stroke-first-500 fill-[none] stroke-[1] "
        d="M559.7142857142857,205 C559.7142857142857,225 628.2857142857143,225 628.2857142857143,245"
        id="111-first"
      />
      <defs>
        <marker
          id="through-arrow-111"
          markerHeight="10px"
          markerUnits="strokeWidth"
          markerWidth="10px"
          orient="180deg"
          refX="100"
          refY="0"
          viewBox="0 0 200 200"
        >
          <path
            class="fill-first-500"
            d="M 200,-0.34918858 100,199.65081 0,-0.34918858 Z"
          />
        </marker>
      </defs>
      <path
        class="stroke-first-500 fill-[none] stroke-[1] custom-animation"
        d="M628.2857142857143,245 C628.2857142857143,225 789.6428571428571,225 789.6428571428571,205"
        id="111-second"
        marker-end="url(#through-arrow-111)"
      />
      <path
        class="fill-[none] stroke-[15] stroke-transparent"
        d="M628.2857142857143,245 C628.2857142857143,225 789.6428571428571,225 789.6428571428571,205"
        id="111-second-wider"
      />
      <text>
        <textpath
          href="#111-first"
          startOffset="0"
          style="font-size: 12px;"
        >
          through
        </textpath>
      </text>
    </svg>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`<ThroughEdge /> if the mouse hovers on the edge renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <svg>
        <path
          class="stroke-first-500 fill-[none] stroke-[1] "
          d="M559.7142857142857,205 C559.7142857142857,225 628.2857142857143,225 628.2857142857143,245"
          id="111-first"
          style="stroke: black; stroke-width: 3; stroke-dasharray: 0;"
        />
        <defs>
          <marker
            id="through-arrow-111"
            markerHeight="10px"
            markerUnits="strokeWidth"
            markerWidth="10px"
            orient="180deg"
            refX="100"
            refY="0"
            viewBox="0 0 200 200"
          >
            <path
              class="fill-first-500"
              d="M 200,-0.34918858 100,199.65081 0,-0.34918858 Z"
            />
          </marker>
        </defs>
        <path
          class="stroke-first-500 fill-[none] stroke-[1] custom-animation"
          d="M628.2857142857143,245 C628.2857142857143,225 789.6428571428571,225 789.6428571428571,205"
          id="111-second"
          marker-end="url(#through-arrow-111)"
          style="stroke: black; stroke-width: 3; stroke-dasharray: 0;"
        />
        <path
          class="fill-[none] stroke-[15] stroke-transparent"
          d="M628.2857142857143,245 C628.2857142857143,225 789.6428571428571,225 789.6428571428571,205"
          id="111-second-wider"
        />
      </svg>
    </div>
  </body>,
  "container": <div>
    <svg>
      <path
        class="stroke-first-500 fill-[none] stroke-[1] "
        d="M559.7142857142857,205 C559.7142857142857,225 628.2857142857143,225 628.2857142857143,245"
        id="111-first"
        style="stroke: black; stroke-width: 3; stroke-dasharray: 0;"
      />
      <defs>
        <marker
          id="through-arrow-111"
          markerHeight="10px"
          markerUnits="strokeWidth"
          markerWidth="10px"
          orient="180deg"
          refX="100"
          refY="0"
          viewBox="0 0 200 200"
        >
          <path
            class="fill-first-500"
            d="M 200,-0.34918858 100,199.65081 0,-0.34918858 Z"
          />
        </marker>
      </defs>
      <path
        class="stroke-first-500 fill-[none] stroke-[1] custom-animation"
        d="M628.2857142857143,245 C628.2857142857143,225 789.6428571428571,225 789.6428571428571,205"
        id="111-second"
        marker-end="url(#through-arrow-111)"
        style="stroke: black; stroke-width: 3; stroke-dasharray: 0;"
      />
      <path
        class="fill-[none] stroke-[15] stroke-transparent"
        d="M628.2857142857143,245 C628.2857142857143,225 789.6428571428571,225 789.6428571428571,205"
        id="111-second-wider"
      />
    </svg>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
